master-slave/src/main/java/com/cisco/oss/foundation/cluster/mongo/AsyncMongoClient.java

```diff
@@ -0,0 +1,140 @@
+package com.cisco.oss.foundation.cluster.mongo;
+
+import com.cisco.oss.foundation.cluster.utils.MasterSlaveConfigurationUtil;
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.allanbank.mongodb.Credential.Builder;
+import com.allanbank.mongodb.MongoClientConfiguration;
+import com.allanbank.mongodb.MongoCollection;
+import com.allanbank.mongodb.MongoDatabase;
+import com.allanbank.mongodb.MongoFactory;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Singleton wrapper over the Mongo Client Driver
+ * Created by Yair Ogen (yaogen) on 14/01/2016.
+ */
+public enum AsyncMongoClient {
+
+	INSTANCE;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AsyncMongoClient.class);
+
+	private Logger logger = null;
+	private static final String DATA_CENTER_COLLECTION = "dataCenter";
+	private static final int DB_RETRY_DELAY = 10;
+	private static final String MASTER_SLAVE_COLLECTION = "masterSlave";
+	public AtomicBoolean IS_DB_UP = new AtomicBoolean(false);
+
+	private MongoDatabase database;
+	private MongoCollection dataCenter;
+
+    private MongoCollection masterSlave;
+
+
+
+
+    AsyncMongoClient() {
+
+		logger = LoggerFactory.getLogger(AsyncMongoClient.class);
+
+		try {
+			database = connect();
+		} catch (MissingMongoConfigException e) {
+			throw e;
+		} catch (Exception e) {
+			infiniteConnect();
+		}
+
+		dataCenter	= database.getCollection(DATA_CENTER_COLLECTION);
+		masterSlave		= database.getCollection(MASTER_SLAVE_COLLECTION);
+	}
+
+
+	private MongoDatabase connect(){
+		MongoClientConfiguration config = new MongoClientConfiguration();
+		List<Pair<String, Integer>> mongodbServers = MasterSlaveConfigurationUtil.getMongodbServers();
+		for (Pair<String, Integer> mongodbServer : mongodbServers) {
+			config.addServer(mongodbServer.getLeft() + ":" + mongodbServer.getRight());
+		}
+    	config.setMaxConnectionCount(10);
+
+    	String dbName = MasterSlaveConfigurationUtil.getMongodbName();
+    	
+		if (MasterSlaveConfigurationUtil.isMongoAuthenticationEnabled()) {
```

Comment:
then user will provide a ready made client using the public connect(MongoClient mongoClient) API